### PR TITLE
689 default context

### PR DESCRIPTION
**Pull request by Jesus89**
Related to https://github.com/CartoDB/cartoframes/issues/689

### CODE REVIEW

File: cartoframes/auth/__init__.py [hunk 1]
```diff
@@ -1,7 +1,18 @@
...
+
+
+def set_default_context(context=None, base_url=None, api_key=None, creds=None, session=None):
+    global default_context
+    if isinstance(context, Context):
+        default_context = context
+    elif isinstance(base_url, str) and isinstance(api_key, str):
+        default_context = Context(base_url, api_key, creds, session)
+    else:
+        raise ValueError('Wrong context data')
```
[the remark this answers is not in the document]
Comment (Jesus89) [reply cartoframes/auth/__init__.py]: OK. But in this case we should include an option to load only the creds if base_url=='' and api_key=='', right? Should we pass the session in both cases?

[the remark this answers is not in the document]
Comment (andy-esch) [reply cartoframes/auth/__init__.py]: Yeah makes sense to only load credentials if `base_url` and `api_key` are not both defined but creds is.

On `session`, we should always pass the session if present. A sessions obj is used with on prems where connection details like ssl verification need to be configured.

[the remark this answers is not in the document]
Comment (Jesus89) [reply cartoframes/auth/__init__.py]: Aha so this should be the code, right?

```py
if isinstance(context, Context):
    default_context = context
elif isinstance(base_url, str) and isinstance(api_key, str):
    default_context = Context(base_url=base_url, api_key=api_key, session=session)
elif isinstance(creds, Credentials):
    default_context = Context(creds=creds, session=session)
```

[the remark this answers is not in the document]
Comment (andy-esch) [reply cartoframes/auth/__init__.py]: Yep, that looks perfect! I'm not feeling sharp today, sorry for not catching this earlier 😥

[the remark this answers is not in the document]
Comment (Jesus89) [reply cartoframes/auth/__init__.py]: :sparkles:

**Pull request merged**